src/glog/logging.h.in

```diff
@@ -649,6 +649,13 @@ void MakeCheckOpValueString(std::ostream* os, const signed char& v);
 template <> GOOGLE_GLOG_DLL_DECL
 void MakeCheckOpValueString(std::ostream* os, const unsigned char& v);
 
+// This is required because nullptr is only present in c++ 11 and later.
+#if __cplusplus >= 201103L
```

Comment:
Unfortunately, this check is unreliable (e.g., under MSVC). Instead, you should test for `std::nullptr_t` using CMake and `CheckCXXSymbolExists` and provide a define (e.g., `HAVE_NULLPTR_T`) that is eventually used here. The `__cplusplus` check can be left as a backup.

Reply:
What about doing like in tensorflow ?
https://github.com/tensorflow/tensorflow/blob/590d6eef7e91a6a7392c8ffffb7b58f2e0c8bc6b/tensorflow/core/platform/macros.h#L108
https://github.com/tensorflow/tensorflow/blob/590d6eef7e91a6a7392c8ffffb7b58f2e0c8bc6b/tensorflow/core/platform/default/logging.h#L171

Reply:
This can be used as fallback. The actual check should be performed by the build system as already explained.